motionplan/inverseKinematics.go

```diff
@@ -2,23 +2,182 @@ package motionplan
 
 import (
 	"context"
+	"encoding/json"
+	"sort"
+
+	"github.com/edaniels/golog"
+	"go.viam.com/utils"
 
 	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
 	spatial "go.viam.com/rdk/spatialmath"
 )
 
 // InverseKinematics defines an interface which, provided with a goal position and seed inputs, will output all found
 // solutions to the provided channel until cancelled or otherwise completes.
-type InverseKinematics interface {
-	// Solve receives a context, the goal arm position, and current joint angles.
-	Solve(context.Context, chan<- []referenceframe.Input, spatial.Pose, []referenceframe.Input, Metric, int) error
+type InverseKinematicsSolver interface {
+	solve(context.Context, chan<- []referenceframe.Input, spatial.Pose, []referenceframe.Input, Metric, int) error
+	frame() referenceframe.Frame
+	options() *ikOptions
+}
+
+type ikSolver struct {
+	model  referenceframe.Frame
+	logger golog.Logger
+	opts   *ikOptions
+}
+
+func (ik *ikSolver) frame() referenceframe.Frame {
+	return ik.model
 }
 
-func limitsToArrays(limits []referenceframe.Limit) ([]float64, []float64) {
-	var min, max []float64
-	for _, limit := range limits {
-		min = append(min, limit.Min)
-		max = append(max, limit.Max)
+func (ik *ikSolver) options() *ikOptions {
+	return ik.opts
+}
+
+func NewIKSolver(frame referenceframe.Frame, logger golog.Logger, ikConfig map[string]interface{}) (InverseKinematicsSolver, error) {
```

Comment:
This is a weird data flow.

The user is expected to manually create a Solver using this function, but can't actually use it (all member methods are private). They then have to pass this into BestIKSolutions.

Rather than having this two-step process, we should just fold this into the `BestIKSolutions` call.

Reply:
Yeah I like this better too.  Will change